### PR TITLE
Ignore cache in occ ldap:check-ldap command

### PR DESCRIPTION
**Pull request by come-nc**
This avoids having to wait or reset the cache after deleting a user in the LDAP.
This also fixes a PHP error when running ldap:check-ldap --update on a deleted but cached user.

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

File: apps/user_ldap/lib/User_LDAP.php [hunk 1]
```diff
@@ -389,13 +390,24 @@ public function deleteUser($uid) {
...
+		$marked = (int)$this->ocConfig->getUserValue($uid, 'user_ldap', 'isDeleted', 0);
+		if ($marked === 0) {
+			try {
+				$user = $this->access->userManager->get($uid);
+				if ($user instanceof User) {
+					$user->markUser();
+					$marked = 1;
+				}
+			} catch (\Exception $e) {
+				// ignore
```
Comment (PVince81) [inline apps/user_ldap/lib/User_LDAP.php:402]: should we log this, at least with debug ?

**Review by PVince81 — APPROVED** [review]
:+1: makes sense, see minor comment

good to see that even when we ignore the cache, we update the value there after reading it

File: apps/user_ldap/lib/User_LDAP.php (hunk 1 above)
Comment (come-nc) [reply apps/user_ldap/lib/User_LDAP.php]: Added debug log of the exception, added it as well in User_Proxy where there is the same try/catch

**Review by juliusknorr — APPROVED** [review]

**Pull request merged**